### DISCUSSION TITLE
Problem with deploying hostedgpt on Render

### DISCUSSION THREAD

**Discussion by ajiewibowo** [AllYourBot/hostedgpt#530] [General]
Hi I tried to deployed hostedgpt on Render as the youtube video instruction but seems like it's failed.
Here are some logs from Render, let me know if there's something I need to change?

#33 exporting to docker image format
#33 exporting layers
#33 exporting layers 29.5s done
#33 exporting manifest sha256:5530d683f87a333eef1e827167e6adf38adba64e65e493feac2d83a0de3e25cf 0.0s done
#33 exporting config sha256:2cd3a95ad510754b667f14bcec4184b4601fbda7eabcc873f35af740727075ea done
#33 DONE 35.0s
#34 exporting cache to client directory
#34 preparing build cache for export
#34 writing cache manifest sha256:a28c5435a482c20d0762aad8c2e0ca70b83ce3b4b422b512d6798995a3d62231 done
#34 DONE 5.9s
Pushing image to registry...
Upload succeeded
==> Deploying...
./bin/rails
Running db:prepare
ERROR: Please set the APP_URL_PROTOCOL environment variable or secret
==> Exited with status 1
==> Common ways to troubleshoot your deploy: https://docs.render.com/troubleshooting-deploys
./bin/rails
Running db:prepare
ERROR: Please set the APP_URL_PROTOCOL environment variable or secret

**Comment by keithschacht** [comment]
Oh shoot, sorry we made a change yesterday. We’ll get it fixed shortly and then we’ll write back here.

@jlvallelonga this is because of yesterdays change, we need to update the render.yml file with the environment variables. It’s worth searching the full codebase for RUN_SOLID_QUEUE_IN_PUMA to see if there are any other places we are setting it and then add the new config. But… actually now that I think about it, can we have some default value within options.yml for the new vars if they are not set? That would be ideal since the various hosting services auto generate url names upon deploy so it’s a bit tricky to set them.

**Comment by keithschacht** [comment]
@ajiewibowo I just rolled back the recent change to main that caused this issue. Sorry for the trouble! If you pull down the latest main everything should work now. I just tested a fresh deploy on Heroku, Render, and Fly and all three worked smoothly this time.
